docker_explorer/de.py

```diff
@@ -85,15 +82,24 @@ def DetectStorage(self):
     if os.path.isfile(
         os.path.join(self.docker_directory, 'repositories-aufs')):
       # Handles Docker engine storage versions 1.9 and below.
-      return aufs.AufsStorage(
+      self.storage_object = aufs.AufsStorage(
           docker_directory=self.docker_directory, docker_version=1)
     elif os.path.isdir(os.path.join(self.docker_directory, u'overlay2')):
-      return overlay.Overlay2Storage(docker_directory=self.docker_directory)
+      self.storage_object = overlay.Overlay2Storage(
+          docker_directory=self.docker_directory)
     elif os.path.isdir(os.path.join(self.docker_directory, 'overlay')):
-      return overlay.OverlayStorage(docker_directory=self.docker_directory)
+      self.storage_object = overlay.OverlayStorage(
+          docker_directory=self.docker_directory)
     elif os.path.isdir(os.path.join(self.docker_directory, 'aufs')):
-      return aufs.AufsStorage(docker_directory=self.docker_directory)
-    return None
+      self.storage_object = aufs.AufsStorage(
+          docker_directory=self.docker_directory)
+    if self.storage_object is None:
+      err_message = (
+          'Could not detect storage system. '
+          'Make sure the docker directory ({0:s}) is correct. '
```

Comment:
Maybe use a more explicit warning? "make sure the docker directory () contains one of [repositories-aufs, overlay, overlay2, aufs]".

Reply:
I'll totally change how storage is detected in a following PR =)